tests/admin_views/test_password_form.py

```diff
@@ -0,0 +1,73 @@
+from django.contrib.admin.tests import AdminSeleniumTestCase
+from django.contrib.auth.models import User
+from django.test import override_settings
+from django.urls import reverse
+
+
+class SeleniumAuthTests(AdminSeleniumTestCase):
+    available_apps = AdminSeleniumTestCase.available_apps
+
+    def setUp(self):
+        self.superuser = User.objects.create_superuser(
+            username="super",
+            password="secret",
+            email="super@example.com",
+        )
+
+    @override_settings(ROOT_URLCONF="auth_tests.urls_admin")
+    def test_unusable_password_javascript(self):
+        """Selecting and deselecting the unsuable password field shows/hides the
+        fields and the warning"""
+        from selenium.common import StaleElementReferenceException
+        from selenium.webdriver.common.by import By
+
+        user_add_url = reverse("auth_test_admin:auth_user_add")
+        self.admin_login(
+            username="super",
+            password="secret",
+        )
+
+        self.selenium.get(self.live_server_url + user_add_url)
+
+        pw_switch_on = self.selenium.find_element(
+            By.CSS_SELECTOR, 'input[name="usable_password"][value="True"]'
+        )
+        pw_switch_off = self.selenium.find_element(
+            By.CSS_SELECTOR, 'input[name="usable_password"][value="False"]'
+        )
+        password1 = self.selenium.find_element(
+            By.CSS_SELECTOR, 'input[name="password1"]'
+        )
+        password2 = self.selenium.find_element(
+            By.CSS_SELECTOR, 'input[name="password2"]'
+        )
+
+        # For add user the default is a usable password
+        self.assertTrue(pw_switch_on.is_selected())
+        self.assertFalse(pw_switch_off.is_selected())
+
+        # Hence the password fields should be visible
+        self.assertTrue(password1.is_displayed())
+        self.assertTrue(password2.is_displayed())
+
+        # Switch usable_password selector
+        pw_switch_off.click()
+
+        # Check the click did switch the checkbox
+        self.assertFalse(pw_switch_on.is_selected())
+        self.assertTrue(pw_switch_off.is_selected())
+
+        # Hence the password fields should be hidden now
+        self.assertFalse(password1.is_displayed())
+        self.assertFalse(password2.is_displayed())
+
+        # Finally check that the warning is displayed
+        warning = self.selenium.find_element(By.ID, "id_unusable_warning")
+        self.assertTrue(warning.is_displayed())
+
+        # Switch usable_password selector once more
+        pw_switch_on.click()
+
+        # Warning disappears
+        with self.assertRaises(StaleElementReferenceException):
+            self.assertFalse(warning.is_displayed())
```

Comment:
This test is great, we'd need a couple more for:
* existing user starting with usable password
* existing user starting with non usable password

Reply:
Done.